src/electron/routing_service.ts

```diff
@@ -254,11 +267,10 @@ function installWindowsRoutingServices(): Promise<void> {
 async function installLinuxRoutingServices(): Promise<void> {
   const OUTLINE_PROXY_CONTROLLER_PATH = path.join('tools', 'outline_proxy_controller', 'dist');
   const LINUX_INSTALLER_FILENAME = 'install_linux_service.sh';
-  // [InstallationFileName, isExecutable]...
-  const LINUX_SERVICE_FILE_NAMES: Array<[string, boolean]> = [
-    [LINUX_INSTALLER_FILENAME, true],
-    ['OutlineProxyController', true],
-    ['outline_proxy_controller.service', false],
+  const LINUX_SERVICE_FILE_DESCRIPTORS: Array<{filename: string; executable: boolean; sha256: string}> = [
```

Comment:
It's unexpected that this is marked as const, and named as const, but you modify the values. See my comment below, you could just remove the sha256 field if you calculate and verify inside the loop.

Reply:
Will change the name, for the verification logic I have responded below.